### PR TITLE
PR: Remove BaseTestCase from `testsuite/`

### PR DESCRIPTION
**Pull request by bigtedde**
Now that the `ArchiverBaseTestCase` has been removed from `testsuite/archiver/__init__.py`, I would now like to remove `BaseTestCase` from the `testsuite/__init__.py`. After @ThomasWaldmann's recent update to the `BaseTestCase`[here](https://github.com/borgbackup/borg/pull/7736), all that remains are a few assert methods that seem superfluous and could be replaced with the built-in python asserts.

### CODE REVIEW

**Comment by bigtedde** [comment]
@ThomasWaldmann There are four test files remaining in the `testsuite` folder that use `BaseTestCase`. One is `/platform.py`, which uses BaseTestCase and a lot of unittest, so I was considering making this its own PR. The other three are files that include tests used in `selftest`, these are:
1. `/hash index.py`
2. `/crypto.py`
3. `/chunker.py`

I feel that removing the BaseTestCase from these would be straightforward and shouldn't impact the selftest themselves as I would not be introducing any pytest or changing any functionality. However, I wanted to check with you first to see if you are fine with me doing this.

File: src/borg/testsuite/helpers.py [hunk 1]
```diff
@@ -396,37 +395,35 @@ def test_chunkerparams():
...
-        self.assert_equal(make_path_safe(".test"), ".test")
-        self.assert_equal(make_path_safe(".test."), ".test.")
-        self.assert_equal(make_path_safe("..test.."), "..test..")
-        self.assert_equal(make_path_safe("/te..st/foo/bar"), "te..st/foo/bar")
-        self.assert_equal(make_path_safe("/..test../abc//"), "..test../abc")
-
-        for path in rejected_dotdot_paths:
-            with pytest.raises(ValueError, match="unexpected '..' element in path"):
-                make_path_safe(path)
+def test_remove_dot_prefixes():
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/helpers.py:398]: the original name had **dotdot**.

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]
looks good. some more places could use parametrize I guess.

**Comment by bigtedde** [comment]
> looks good. some more places could use parametrize I guess.

I will make that happen. Also, are you okay with me removing BaseTestCase from the 3 files in `selftest`? Being careful to not change any functionality and not introducing pytest.

**Comment by ThomasWaldmann** [comment]
Maybe stay away from the selftest stuff until for now until the other stuff is done.

File: src/borg/testsuite/helpers.py (hunk 1 above)
Comment (ThomasWaldmann) [reply src/borg/testsuite/helpers.py:398]: seen this?

Comment (bigtedde) [reply src/borg/testsuite/helpers.py:398]: Yes, I've got this corrected locally. Adding some additional parameterization to this file and will push soon 🙂

**Comment by bigtedde** [comment]
Cleaned up and parametrized as needed in `helpers.py`

File: src/borg/testsuite/helpers.py [hunk 2]
```diff
@@ -14,39 +14,54 @@
...
+    remove_dotdot_prefixes,
     replace_placeholders,
+    safe_ns,
+    safe_s,
+    safe_unlink,
+    swidth_slice,
+    text_to_json,
+    text_validator,
+    yes,
 )
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/helpers.py:62]: That stuff uses to be grouped (more or less what belongs together somehow).

Or did you somehow trigger black doing this?

I rather prefer the horizontal grouped form, less scrolling.

**Review by ThomasWaldmann — COMMENTED** [review]
Tests look much better with parametrize now.

Just the imports somehow got ugly, IMHO.

File: src/borg/testsuite/helpers.py (hunk 2 above)
Comment (bigtedde) [reply src/borg/testsuite/helpers.py:62]: I'll get this changed back to the original 🙂

**Comment by bigtedde** [comment]
@ThomasWaldmann As far as removing BaseTestCase from the `selftest` items - is this something I'd be good to work on now, or should we hold off?

**Review by ThomasWaldmann — APPROVED** [review]
LGTM.

**Pull request merged**